OurUmbraco.Site/Views/Partials/Forum/Forum.cshtml

```diff
@@ -168,19 +171,22 @@
         <nav class="pagination" role="navigation">
             @if (page > 1)
             {
-                <a class="prev" href="?page=@(page - 1)">Prev</a>
+                queryString["page"] = (page - 1).ToString();
+                <a class="prev" href="?@queryString">Prev</a>
                 <span>&hellip;</span>
             }
 
             @for (var i = (page - 1 > 0 ? page - 1 : 1); i < (page + (pagesToShowLeft - (page - 1 > 0 ? 1 : 0))) && i <= pages; i++)
             {
-                <a class="@Umbraco.If(i == page, "active")" href="?page=@i">@i</a>
+                queryString["page"] = i.ToString();
+                <a class="@Umbraco.If(i == page, "active")" href="?@queryString">@i</a>
             }
 
-            @if (page < (pages - 3))
+            @if (page < pages)
             {
+                queryString["page"] = (page + 1).ToString();
                 <span>&hellip;</span>
-                <a class="next" href="?page=@(page + 1)">Next</a>
+                <a class="next" href="?@queryString">Next</a>
```

Comment:
Hi Shannon,

Thanks for the feedback. Did you have a look at this?

Forgive me if I'm being ignorant but won't ASP.NET protect against this? First of all any script tags in the query string will cause an `HttpRequestValidationException` ("A potentially dangerous Request.QueryString value was detected from the client"). Second, Razor will automatically encode strings output with `@` (apart from `HtmlString` of course but that's not what we're dealing with here).

I've attempted to execute an XSS attack but without any luck. Is there something I'm missing?

Cheers,

Steven